addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1028,19 +1038,17 @@ var SnippetEditor = Widget.extend({
                 $selectorChildren = $selectorChildren.add(self.selectorChildren[i].all());
             }
         }
-        // TODO In master, do not reference other module class + find a better
-        // system to define such cases + avoid duplicated code (drag & drop from
-        // editor panel + drag & drop from move button of existing block).
-        // Prevent dropping ToC inside another ToC. grep: NO_DOUBLE_TOC
-        if (this.$target[0].classList.contains('s_table_of_content')) {
-            $selectorChildren = $selectorChildren.filter((i, el) => !el.closest('.s_table_of_content'));
-        }
-        // Disallow dropping form fields outside of their form.
-        // TODO this can probably be implemented by reviewing data-drop-near
-        // definitions in master but we should find a better to define those and
-        // such cases.
-        if (this.$target[0].classList.contains('s_website_form_field')) {
-            const filterFunc = (i, el) => el.closest('form') === closestFormEl;
+        // Prevent dropping an element into another one.
+        // (E.g. ToC inside another ToC)
+        for (const excludedParentSelector of this.selectorExcludeParent) {
+            $selectorSiblings = $selectorSiblings?.filter((i, el) => !el.closest(excludedParentSelector));
+            $selectorChildren = $selectorChildren?.filter((i, el) => !el.closest(excludedParentSelector));
```

Comment:
- Those lines are too long
- Now we are doing
```
var $selectorSiblings;
for (...) {
    // Add things in $selectorSiblings;
}
var $selectorChildren;
for (...) {
    // Add things in $selectorChildren;
}
for (...) {
    // Filter to removes elements from $selectorChildren and $selectorSiblings
}
```
I wonder if you could just do a check when we add stuff in `$selectorSiblings` and in `$selectorChildren` instead of filter after. :thinking:

Reply:
Did this: https://github.com/odoo/odoo/compare/88c042f42ab482cbcd4025ad6209b4b731b79732..4a369a2fa07fad295ae2d5c31e7a23c113b76615>